File: exercism/javascript/complete/secret-handshake/secret-handshake.js
const secretHandshake = (hex) => {
  const regex = /^[0-9]*$/g
  if (!regex.test(hex)) throw(new Error('Handshake must be a number'));
  let output = []
  const hash = { 1:"wink", 2:"double blink", 4:"close your eyes", 8:"jump", 16:"reverse" }

  for (let key in hash) {
    if (hex == 0) break
    if ((hex & key) == key) {
      key == 16 ? output.reverse() : output.push(hash[key])
      hex -= key
    }
  }

  return output
}

export { secretHandshake }

File: exercism/ruby/complete/secret-handshake/secret_handshake.rb
class SecretHandshake
  @@secrets = {
    'wink': 0b1,
    'double blink': 0b10,
    'close your eyes': 0b100,
    'jump': 0b1000,
    'reverse': 0b10000
  }

  def initialize decimal
    @decimal = decimal
  end

  def commands
    output = []
    return output unless @decimal.is_a? Integer
    @@secrets.each do |key, binary|
      if @decimal & binary == binary
        if binary == 0b10000
          output.reverse!
        else
          output << key.to_s
          @decimal ^= binary
        end
      end
    end
    output
  end
end
